src/main/java/com/liftlife/liftlife/dietModule/DietController.java

```diff
@@ -0,0 +1,114 @@
+package com.liftlife.liftlife.dietModule;
+
+import com.liftlife.liftlife.dietModule.dietDay.DietDay;
+import com.liftlife.liftlife.dietModule.dietDay.DietDayRepository;
+import com.liftlife.liftlife.dietModule.dietDay.meal.Meal;
+import com.liftlife.liftlife.dietModule.dietPlan.DietPlan;
+import com.liftlife.liftlife.util.database.FirestoreEntity;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/api/dietDay")
+public class DietController {
+
+    private final DietService dietService;
+
+    @Autowired
+    public DietController(DietService dietService) {
+        this.dietService = dietService;
+    }
+
+    @GetMapping("/meals/{dietId}/{mealId}")
+    public ResponseEntity<Meal> getMealById(@PathVariable String dietId, @PathVariable String mealId) {
+        Meal meal = dietService.findById(dietId, mealId);
+        return ResponseEntity.ok(meal);
+    }
+
+    @GetMapping("/{type}/{id}")
+    public <T extends FirestoreEntity> ResponseEntity<T> getById(@PathVariable String type, @PathVariable String id) {
+        DietServiceType serviceType = DietServiceType.valueOf(type.toUpperCase());
+        T entity = dietService.findById(id, serviceType);
+        return ResponseEntity.ok(entity);
+    }
+
+    @GetMapping("/meals/name/{dietId}")
+    public ResponseEntity<List<Meal>> getMealsByName(@PathVariable String dietId, @RequestParam String name) {
+        List<Meal> meals = dietService.findByName(dietId, name);
+        return ResponseEntity.ok(meals);
+    }
+
+    @GetMapping("/meals/{dietId}")
+    public ResponseEntity<List<Meal>> getMealsByFields(@PathVariable String dietId, @RequestParam Map<String, Object> fields) {
```

Comment:
Imo that endpoint may be too complex and generic for usage on frontend side because requires knowledge of every field on class. I'd ask guys from frontend whether it's acceptable, if they approve, I do too.